src/meta/src/barrier/mod.rs

```diff
@@ -1389,6 +1353,65 @@ impl CheckpointControl {
     pub(super) async fn next_completed_barrier(
         &mut self,
     ) -> MetaResult<Option<BarrierCompleteOutput>> {
+        if matches!(&self.completing_command, CompletingCommand::None) {
+            // `Vec::new` is a const fn, and do not have memory allocation, and therefore is lightweight enough
+            let mut finished_jobs = Vec::new();
+            let min_upstream_inflight_barrier = self
+                .command_ctx_queue
+                .first_key_value()
+                .map(|(epoch, _)| *epoch);
+            for (table_id, job) in &mut self.creating_streaming_job_controls {
+                if let Some((epoch, resps, status)) =
+                    job.start_completing(min_upstream_inflight_barrier)
+                {
+                    let is_first_time = match status {
+                        CompleteJobType::First => true,
+                        CompleteJobType::Normal => false,
+                        CompleteJobType::Finished => {
+                            finished_jobs.push((*table_id, epoch, resps));
+                            continue;
+                        }
+                    };
+                    let tables_to_commit = job
+                        .info
+                        .table_fragments
+                        .all_table_ids()
+                        .map(TableId::new)
+                        .collect();
+                    let join_handle =
+                        tokio::spawn(self.context.clone().complete_creating_job_barrier(
+                            epoch,
+                            resps,
+                            tables_to_commit,
+                            is_first_time,
+                        ));
+                    self.completing_command = CompletingCommand::CreatingStreamingJob {
```

Comment:
IIUC, currently we are polling commands for the following barriers here:
1. Collected fake barrier for creating jobs
2. Collected normal barrier for createing jobs
3. Collected normal barrier for created jobs

1 and 2 are biased over 3. Is it possible that this can cause starvation in commit_epoch for 3 and thus delay the regular checkpoint? For example, right before each time `next_completed_barrier` is called, there is a barrier from 1 or 2 finishing its collection. In the worst case, regular ckpt cannot be completed until the new jobs finish creation.

Reply:
Will not have starvation after https://github.com/risingwavelabs/risingwave/pull/18819